pkg/defaults/defaults.go

```diff
@@ -108,36 +123,60 @@ func deconstructBlueprintsToCreationSteps(rawBlueprints []port.Blueprint) ([]por
 
 type AbortDefaultCreationError struct {
 	BlueprintsToRollback []string
+	PagesToRollback      []string
 	Errors               []error
 }
 
 func (e *AbortDefaultCreationError) Error() string {
 	return "AbortDefaultCreationError"
 }
 
-func validateBlueprintErrors(createdBlueprints []string, blueprintErrors []error) *AbortDefaultCreationError {
-	if len(blueprintErrors) > 0 {
-		for _, err := range blueprintErrors {
+func validateResourcesErrors(createdBlueprints []string, createdPages []string, resourceErrors []error) *AbortDefaultCreationError {
+	if len(resourceErrors) > 0 {
+		for _, err := range resourceErrors {
 			klog.Infof("Failed to create resources: %v.", err.Error())
 		}
-		return &AbortDefaultCreationError{BlueprintsToRollback: createdBlueprints, Errors: blueprintErrors}
+		return &AbortDefaultCreationError{BlueprintsToRollback: createdBlueprints, PagesToRollback: createdPages, Errors: resourceErrors}
 	}
 	return nil
 }
 
-func createResources(portClient *cli.PortClient, defaults *Defaults, config *port.Config) *AbortDefaultCreationError {
+func validateResourcesDoesNotExist(portClient *cli.PortClient, defaults *Defaults, config *port.Config) *AbortDefaultCreationError {
+	var errors []error
 	if _, err := integration.GetIntegration(portClient, config.StateKey); err == nil {
-		return &AbortDefaultCreationError{Errors: []error{
-			fmt.Errorf("integration with state key %s already exists", config.StateKey),
-		}}
+
+	}
+
+	for _, bp := range defaults.Blueprints {
+		if _, err := blueprint.GetBlueprint(portClient, bp.Identifier); err == nil {
+			errors = append(errors, fmt.Errorf("blueprint with identifier %s already exists", bp.Identifier))
+		}
+	}
+
+	for _, p := range defaults.Pages {
+		if _, err := page.GetPage(portClient, p.Identifier); err == nil {
+			errors = append(errors, fmt.Errorf("page with identifier %s already exists", p.Identifier))
+		}
+	}
+
+	if errors != nil {
+		return &AbortDefaultCreationError{Errors: errors}
+	}
+	return nil
+}
+
+func createResources(portClient *cli.PortClient, defaults *Defaults, config *port.Config) *AbortDefaultCreationError {
```

Comment:
There is some scale issue with the current code architecture. One or two more resources and this function and the entire file will be too huge to maintain. Up to you if to solve it now

Reply:
We plan to handle it in port api so i think its ok for now